Remove early return from gizmo intersect_ray

AndreaCatania opened the pull request:
Remove early return from `EditorNode3DGizmo::intersect_ray` that is preventing to have gizmos that use Mesh collision + Segment collision.

JFonS commented:
Actually, thinking about it, we might as well remove the return in the selectable icon check some lines above: https://github.com/godotengine/godot/blob/659178e85fed5a7ee80d76329a977ac3943c7d77/editor/node_3d_editor_gizmos.cpp#L604

Otherwise we will have the same problem with gizmos that have both an icon and selectable segments or meshes.

AndreaCatania commented:
Yee, I agree, I can remove that as well.

AndreaCatania committed:
Remove early return from gizmo intersect_ray

Remove early returns from `EditorNode3DGizmo::intersect_ray` that is preventing to have gizmos that use Mesh collision + Segment collision + Icon.

AndreaCatania commented:
Done

akien-mga commented:
Thanks!

akien-mga commented:
Cherry-picked for 3.4.
